File: OOP/Polymorphism.py
import math


class Shape:
    def area(self):
        return 0

    def __str__(self):
        return f"Shape with area {self.area():.2f}"


class Rectangle(Shape):
    def __init__(self, w, h):
        self.w = w
        self.h = h

    def area(self):
        return self.w * self.h


class Circle(Shape):
    def __init__(self, r):
        self.r = r

    def area(self):
        return math.pi * self.r ** 2


shapes = [Rectangle(4, 5), Circle(3)]
for s in shapes:
    print(str(s))

File: OOP/Polymorphism.js
class Shape {
  area()      { return 0; }
  toString()  { return `Shape with area ${this.area().toFixed(2)}`; }
}

class Rectangle extends Shape {
  constructor(w, h) { super(); this.w = w; this.h = h; }
  area() { return this.w * this.h; }
}

class Circle extends Shape {
  constructor(r) { super(); this.r = r; }
  area() { return Math.PI * this.r ** 2; }
}

const shapes = [new Rectangle(4, 5), new Circle(3)];
shapes.forEach(s => console.log(s.toString()));
// "Shape with area 20.00"
// "Shape with area 28.27"